public/scripts/merchandise.js

```diff
@@ -0,0 +1,405 @@
+		$("#merchandise-data").empty();
+
+		snap.forEach(function (childSnap) {
+			
+			var merchandiseId = childSnap.key;
+			var merchandiseClickUrl = childSnap.child("onClickUrl").val();
+			var merchandiseImage = childSnap.child("imageUrl").val();
+			var merchandiseName = childSnap.child("name").val();
+			var merchandisePrice = childSnap.child("price").val();
+			var merchandiseDescription = childSnap.child("desc").val();
+
+			merchandiseEditData[merchandiseId] = {
+				merchandiseClickUrl : merchandiseClickUrl,
+				merchandiseImage : merchandiseImage,
+				merchandiseName : merchandiseName,
+				merchandisePrice : merchandisePrice,
+				merchandiseDescription : merchandiseDescription,
+				fileLoc : null
+
+			};
+
+			var template = "<div id='"+merchandiseId+"' class='col-lg well-lg bg-1 div-mod '>"
+			+"<img src='"+merchandiseImage+"' class='img-responsive img-1 alt='"+merchandiseName+"' '><br>"
+			+"<input type='button' class='btn btn-primary' value='Edit' onclick='editMerchandise(this.parentNode.id,$(this))'>"
+			+"<h3>Merchanise Name:<span class='merchandise-name'>"+merchandiseName+"</span></h3>"
+			+"<h3>Merchanise Description:<span class='merchandise-description'>"+merchandiseDescription+"</span></h3>"
+			+"<h3>Merchanise Price:<span class='merchandise-price'>"+merchandisePrice+"</span></h3>"
+			+"<h3>Merchanise Link:<span class='merchandise-link'>"+merchandiseClickUrl+"</span></h3>"
+			+"</div>";
+			
+			$("#merchandise-data").append($(template));
+
+		});
+			
+		/*This code must execute only the first time the value event triggers */
+		if(firstTime){
+			$("#add-new-merchandise").hide();
+			var t=1;
+			$("#add-merchandise-btn").click(function () {			
+
+				if(t) {
+
+					$(this).val("Cancel");
+					
+				} else {
+					$(this).val("Add New Merchanise");
+				}
+				t^=1;
+				$("#add-new-merchandise").fadeToggle();						
+			});
+			$("#add-new-div").show();
+			firstTime = false;	
+		}
+	},
+		function error(dberror) {
+			console.error(dberror.message);
+		}
+	);
+	
+
+});
+
+function editMerchandise(parentId,btnRef) {
+
+	
+	$originalTemplate = $("#"+parentId).clone();
+	$("#" +parentId+" .merchandise-name").replaceWith($("<input class='merchandise-name' type='text' value='"+
+							$("#"+parentId+" .merchandise-name").text()+"' >"));
+	$("#" +parentId+" .merchandise-link").replaceWith($("<input class='merchandise-link' type='text' value='"+
+							$("#"+parentId+" .merchandise-link").text()+"' >"));
+
+	$("#" +parentId+" .merchandise-price").replaceWith($("<input class='merchandise-price' type='text' value='"+
+							$("#"+parentId+" .merchandise-price").text()+"' >"));
+
+	$("#" +parentId+" .merchandise-description").replaceWith($("<input class='merchandise-description' type='text' value='"+
+							$("#"+parentId+" .merchandise-description").text()+"' >"));
+	btnRef.val("Save");
+	$("#"+parentId+" .image").after($("<br><input type='button' class='btn btn-secondary' value='Change Image'><br>")
```

Comment:
Change .image to img since the referrence is to the tag.
This will fix the Change Image btn